src/filesystem.c

```diff
@@ -238,56 +238,185 @@ rcutils_mkdir(const char * abs_path)
   return success;
 }
 
-size_t
+ssize_t
 rcutils_calculate_directory_size(const char * directory_path, rcutils_allocator_t allocator)
 {
-  size_t dir_size = 0;
+  return rcutils_calculate_directory_size_with_recursion(directory_path, 1, allocator);
+}
+
+typedef struct dir_list_t
+{
+  char * path;
+  uint32_t depth;  // Compare with base path
+  struct dir_list_t * next;
+} dir_list_t;
+
+static inline void free_dir_list(dir_list_t * dir_list, rcutils_allocator_t allocator)
```

Comment:
Let's remove the `inline` here (and below).  It is only a hint to the compiler anyway, and it can figure out whether it wants to inline it or not.

```suggestion
static void free_dir_list(dir_list_t * dir_list, rcutils_allocator_t allocator)
```